crates/astria-composer/src/executor/bundle_factory/mod.rs

```diff
@@ -24,17 +33,34 @@ enum SizedBundleError {
     SequenceActionTooLarge(SequenceAction),
 }
 
+pub(super) struct RollupCountsReport(HashMap<RollupId, u32>);
+
+impl Serialize for RollupCountsReport {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(self.0.len()))?;
+        for (rollup_id, count) in &self.0 {
+            map.serialize_entry(rollup_id, count)?;
+        }
+        map.end()
+    }
+}
+
 /// A bundle sequence actions to be submitted to the sequencer. Maintains the total size of the
 /// bytes pushed to it and enforces a max size in bytes passed in the constructor. If an incoming
 /// `seq_action` won't fit in the buffer it is flushed and a new bundle is started.
-struct SizedBundle {
+pub(super) struct SizedBundle {
```

Comment:
Same with all of these fields - why are they no longer module-private? Due to instrumentation? Can you just provide accessors?

Reply:
Changed this `Report` thing to resemble what you did in `conductor` more closely, should get rid of the weird visibility stuff